src/log_outgoing_requests/tests/test_logging.py

```diff
@@ -58,26 +71,40 @@ def test_expected_data_is_saved_when_saving_enabled(self, m):
                 self.assertEqual(request_log.query_params, "version=2.0")
                 self.assertEqual(response.status_code, 200)
                 self.assertEqual(request_log.method, method)
-                self.assertEqual(request_log.req_content_type, "")
-                self.assertEqual(request_log.res_content_type, "")
+                self.assertEqual(request_log.req_content_type, "text/html")
+                self.assertEqual(request_log.res_content_type, "application/json")
                 self.assertEqual(request_log.response_ms, 0)
                 self.assertEqual(
                     request_log.req_headers,
                     (
-                        "{'User-Agent': 'python-requests/2.26.0', 'Accept-Encoding': 'gzip, deflate, br', 'Accept': '*/*', 'Connection': 'keep-alive'}"
-                        if method == "HEAD"
-                        else "{'User-Agent': 'python-requests/2.26.0', 'Accept-Encoding': 'gzip, deflate, br', 'Accept': '*/*', 'Connection': 'keep-alive', 'Content-Length': '0'}"
+                        "User-Agent: python-requests/2.26.0\n"
+                        "Accept-Encoding: gzip, deflate, br\n"
+                        "Accept: */*"
+                        "\nConnection: keep-alive\n"
+                        "Authorization: ***hidden***\n"
+                        "Content-Type: text/html"
+                        if method in ["HEAD", "GET"]
+                        else "User-Agent: python-requests/2.26.0\n"
+                        "Accept-Encoding: gzip, deflate, br\n"
+                        "Accept: */*\n"
+                        "Connection: keep-alive\n"
+                        "Authorization: ***hidden***\n"
+                        "Content-Type: text/html\n"
+                        "Content-Length: 0"
```

Comment:
There is way too much in this if/else expression, and it is very hard to see which parts are the same and what is the difference between each case.

As a general rule if/else expressions are not so great;  for anything non trivial it is almost always more readable to use normal if/else statements.